cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/AnomalyDetectorState.java

```diff
@@ -62,8 +66,21 @@ protected boolean removeEldestEntry(Map.Entry<String, AnomalyState> eldest) {
       });
     }
     _selfHealingEnabled = selfHealingEnabled;
+    _ongoingSelfHealingAnomaly = new AtomicReference<>();
   }
 
+  /**
+   * Update anomaly status once {@link com.linkedin.kafka.cruisecontrol.executor.Executor} has finished corresponding
+   * self-healing operation.
+   *
+   * @param anomalyId UUID of anomaly which triggered self-healing operation.
+   */
+  public void selfHealingFinished(String anomalyId) {
+    if (!anomalyId.equals(_ongoingSelfHealingAnomaly.get().anomalyId())) {
+      throw new IllegalStateException(String.format("Anomaly %s is not marked as FIX_START state in AnomalyDetector.", anomalyId));
+    }
+    _ongoingSelfHealingAnomaly.set(null);
```

Comment:
I think the read is not a concern here since I may get an slightly stale ongoing anomaly but I get a (snapshot-level )consistent value.

I agree that write will be a potential source of bug.My original idea is to use CAS operation of atomic variable, but I cannot do it here since we cannot do compare atomically(need to dereference and compare the anomalyId field).I fall back to `synchronized` due to 
(1) though as you mentioned, current access pattern of `onAnomalyHandle` and `selfHealingFinished` ensures there is no race condition but we cannot guarantee for future
(2)there place is not a heavily contented place, so `synchronized`  will not bring much extra load.